backend/vkt/src/main/java/fi/oph/vkt/service/PublicEnrollmentEmailService.java

```diff
@@ -210,15 +214,73 @@ public void sendFreeEnrollmentToQueueConfirmationEmail(
       LocalisationUtil.translate(localeFI, "subject.enrollment-to-queue-confirmation"),
       LocalisationUtil.translate(localeSV, "subject.enrollment-to-queue-confirmation")
     );
-    final String body = templateRenderer.renderFreeEnrollmentToQueueConfirmationEmailBody(
-      templateParams,
-      enrollment.getFreeEnrollment().getSource()
+    final String body = templateRenderer.renderEnrollmentConfirmationEmailBody(templateParams);
+
+    createEmail(recipientName, recipientAddress, subject, body, List.of(), EmailType.ENROLLMENT_TO_QUEUE_CONFIRMATION);
+  }
+
+  @Transactional
+  public void sendPartiallyFreeEnrollmentConfirmationEmail(
+    final Enrollment enrollment,
+    final Person person,
+    final FreeEnrollmentDetails freeEnrollmentDetails
+  ) throws IOException, InterruptedException {
+    final Map<String, Object> templateParams = getEmailParams(enrollment);
+    getFreeEmailParams(templateParams, freeEnrollmentDetails, enrollment.getFreeEnrollment().getSource(), "queue");
+    templateParams.put("isFree", "false");
+    templateParams.put("type", "enrollment");
```

Comment:
Mielestäni tämän templateParams-manipuloinnin voisi siirtää kokonaan tuonne `getFreeEmailParams`-metodin sisään.

Reply:
Tyyliseikkana myöskin voisi harkita uudelleennimeämistä (esim. `getFreeEmailParams` -> `withFreeEmailParams`) ja mutatoinnin sijaan myös palauttaisin uuden mapin.

Tällöin kutsujalle käyttö voisi typistyä rimpsuun `final Map<String, Object> templateParams = withFreeEmailParams(getEmailParams(enrollment), freeEnrollmentDetails, ....);`